language/move-lang/src/parser/syntax.rs

```diff
@@ -307,6 +307,46 @@ fn parse_module_access<'input, F: FnOnce() -> String>(
     Ok(spanned(tokens.file_name(), start_loc, end_loc, acc))
 }
 
+//**************************************************************************************************
+// Visibility
+//**************************************************************************************************
+
+// Parse a function visibility modifier:
+//      FunctionVisibility = ( "public" ( "(" "script" ")" )? )?
+fn parse_function_visibility<'input>(
+    tokens: &mut Lexer<'input>,
+) -> Result<FunctionVisibility, Error> {
+    let visibility = if tokens.peek() == Tok::Public {
+        let start_loc = tokens.start_loc();
+        tokens.advance()?;
+        let sub_public_vis = if match_token(tokens, Tok::LParen)? {
+            let sub_token = tokens.peek();
+            tokens.advance()?;
+            if sub_token != Tok::RParen {
+                consume_token(tokens, Tok::RParen)?;
+            }
+            Some(sub_token)
+        } else {
+            None
+        };
+        let end_loc = tokens.previous_end_loc();
+        // this loc will cover the span of 'public' or 'public(...)' in entirety
+        let loc = make_loc(tokens.file_name(), start_loc, end_loc);
+        match sub_public_vis {
+            None => FunctionVisibility::Public(loc),
+            Some(Tok::Script) => FunctionVisibility::Script(loc),
+            _ => {
+                let msg = "Invalid visibility modifier. \
+                Consider removing it or using with one of `public` or `public(script)`";
```

Comment:
Drop the "with" in "...using with one of..."

Reply:
Ah... what am i thinking...